swaggen/src/main/java/generator/PathGenerator.java

```diff
@@ -54,9 +54,59 @@
 	 * @param annotation the annotation
 	 * @return the path
 	 */
-	private static Map<RequestMethod, Endpoint> generatePathFromAnnotation(SwaggerGen annotation) {
-		Map<RequestMethod, Endpoint> endpointMap = new HashMap<>();
-		endpointMap.put(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
-		return endpointMap;
+	private static SwaggerEndpoint generatePathFromAnnotation(SwaggerGen annotation) {
+		SwaggerEndpoint endpoint = new SwaggerEndpoint();
+		endpoint.addEndpoint(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
+		return endpoint;
+	}
+
+	/**
+	 * Checks if URL is already in the existing map, and if it is, appends the swagger
+	 * endpoints to the URL
+	 * Case: multiple classes with the same endpoint
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 * @param classPaths Map of new Swagger Endpoints in the class
+	 */
+	private static void checkClassEndpoints(Map<PathURL, SwaggerEndpoint> existingPaths, Map<PathURL, SwaggerEndpoint> classPaths) {
+		if (existingPaths.isEmpty()) {
+			existingPaths.putAll(classPaths);
+			return;
+		}
+
+		for(PathURL pathKey: existingPaths.keySet()) {
+			if (classPaths.containsKey(pathKey)) {
+				existingPaths.get(pathKey).addEndpoint(classPaths.get(pathKey));
+			} else {
+				existingPaths.putAll(classPaths);
+			}
+		}
+	}
+
+	/**
+	 * Checks if there are any other request methods within a class
+	 * and puts it in the swagger endpoints map
+	 * Case: multiple request methods within the same class
+	 * @param annotation SwaggerGen annotation
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 */
+	private static void checkRequestMethods(SwaggerGen annotation, Map<PathURL, SwaggerEndpoint> existingPaths) {
+		if (annotation == null) {
+			throw new IllegalArgumentException("annotation cannot be null");
+		}
+		
+		PathURL pathURL = new PathURL(annotation.basePath(), annotation.uri());
+		if (!existingPaths.containsKey(pathURL)){
+			existingPaths.put(pathURL, generatePathFromAnnotation(annotation));
+		} else {
+			existingPaths.get(pathURL).addEndpoint(generatePathFromAnnotation(annotation));
+		}
+		// for (PathURL paths : existingPaths.keySet()) {

```

Comment:
Don't comment out code, delete it. We have source control.

The only time there should be commented code is if it is for a future feature and it has a clear TODO message describing when it should be uncommented.